tensorflow_probability/python/distributions/johnson_su.py

```diff
@@ -0,0 +1,281 @@
+          loc, name='loc', dtype=dtype)
+      self._scale = tensor_util.convert_nonref_to_tensor(
+          scale, name='scale', dtype=dtype)
+
+      norm_shift = invert_bijector.Invert(
+          shift_bijector.Shift(shift=self._skewness,
+                               validate_args=validate_args)
+      )
+
+      norm_scale = invert_bijector.Invert(
+          scale_bijector.Scale(scale=self._tailweight,
+                               validate_args=validate_args)
+      )
+
+      sinh = sinh_bijector.Sinh(validate_args=validate_args)
+
+      scale = scale_bijector.Scale(scale=self._scale,
+                                   validate_args=validate_args)
+
+      shift = shift_bijector.Shift(shift=self._loc,
+                                   validate_args=validate_args)
+
+      bijector = shift(scale(sinh(norm_scale(norm_shift))))
+
+      batch_shape = distribution_util.get_broadcast_shape(
+          self._skewness, self._tailweight, self._loc, self._scale)
+
+      super(JohnsonSU, self).__init__(
+          distribution=normal.Normal(
+              loc=tf.zeros([], dtype=dtype),
+              scale=tf.ones([], dtype=dtype),
+              validate_args=validate_args,
+              allow_nan_stats=allow_nan_stats),
+          bijector=bijector,
+          batch_shape=batch_shape,
+          validate_args=validate_args,
+          parameters=parameters,
+          name=name)
+
+  @staticmethod
+  def _param_shapes(sample_shape):
+    return dict(
+        zip(('skewness', 'tailweight', 'loc', 'scale'),
+            ([tf.convert_to_tensor(sample_shape, dtype=tf.int32)] * 4)))
+
+  @classmethod
+  def _params_event_ndims(cls):
+    return dict(skewness=0, tailweight=0, loc=0, scale=0)
+
+  @property
+  def skewness(self):
+    """Skewness of these Johnson's SU distribution(s)."""
+    return self._skewness
+
+  @property
+  def tailweight(self):
+    """Tail weight of these Johnson's SU distribution(s)."""
+    return self._tailweight
+
+  @property
+  def loc(self):
+    """Locations of these Johnson's SU distribution(s)."""
+    return self._loc
+
+  @property
+  def scale(self):
+    """Scaling factors of these Johnson's SU distribution(s)."""
+    return self._scale
+
+  def _mean(self):
+    skewness, tailweight, scale, loc = (
+        [tf.convert_to_tensor(v)
+         for v in (self.skewness, self.tailweight, self.scale, self.loc)])
+
+    return (loc - scale * tf.math.exp(0.5 / tf.math.square(tailweight)) *
+            tf.math.sinh(skewness / tailweight))
+
+  def _variance(self):
+    skewness, tailweight, scale = (
+        [tf.convert_to_tensor(v)
+         for v in (self.skewness, self.tailweight, self.scale)])
+
+    variance = (0.5 * tf.math.square(scale) *
+                tf.math.expm1(1./tf.math.square(tailweight)) *
+                (tf.math.exp(1./tf.math.square(tailweight)) *
+                 tf.math.cosh(2. * skewness / tailweight) + 1.))
+
+    return tf.broadcast_to(variance, self.batch_shape_tensor())
```

Comment:
Could you elaborate on this? Do you mean `EXTRA_TENSOR_CONVERSION_DISTS` in distr properties test? It seems to be required for previous implementation, dropped the change and test is passing. Otherwise I'm not sure what can be optimized here. Base class (transformed distr) has it's own `_batch_shape_tensor` impl.

BTW, I've applied your suggestion above and @googlebot  now requires CLA from your side. What can be done here?

Reply:
Sorry I missed this. Can you try signing this again and triggering Google bot?

My recommendation was overriding _batch_shape_tensor, but glad things worked out.

Reply:
I suspect that's because you are mentioned as a co-author in auto-generated commit https://github.com/tensorflow/probability/pull/800/commits/08c2c0abe9cfc89d8531fe349fc785d80d5ff5a6 Probably I can edit commit message (via rebase) and provide your email which google bot is aware of?

Reply:
Hrmm I'll try. I'm already recognized as part of @google.com, but let me check.

Reply:
No reaction from the bot :(

Reply:
If you don't mind I've changed your email in commit message from @gmail.com to @google.com just like it appears in other commits. Googlebot is happy with this now.

Reply:
No worries. That's super weird it even used @gmail in the first place (since that isn't / shouldn't be my primary). Thanks for finding the fix!